app/static/src/app/utils.ts

```diff
@@ -68,13 +66,30 @@ export function processFitData(data: Dict<string>[], errorMsg: string): ProcessF
         const error = { error: errorMsg, detail };
         return { ...emptyResult, error };
     }
+
+    // There might be trailing empty rows to drop.
+    while (processedData.length > 0) {
+        const row = processedData[processedData.length - 1];
+        const missing = Object.values(row).map((v) => Number.isNaN(v));
+        if (missing.reduce((a, b) => a && b, true)) { // all missing
```

Comment:
```suggestion
       if (Object.values(row).every((v) => Number.isNaN(v))) { // all missing
```
I think this is equivalent

Reply:
thanks - my googling failed to find this, but it's exactly what I was looking for